### PR TITLE
[sdl2-gfx] Update CMake build and find_package support

### PR DESCRIPTION
**Pull request by dhkatz**
This allows for using sdl2-gfx easily by generating the proper files (sdl2-gfx-config.cmake and such) when building and providing it to vcpkg-cmake-wrapper.cmake to allow easy `find_package()` usage.

Makes it easy to just `find_package(sdl2-gfx) target_link_libraries(main PRIVATE SDL2::SDL2_gfx)` (as suggested by the porting guidelines) instead of having to mess with other methods.

I tried to follow a similar CMake build process to the SDL2_image port.

Just a fun note, I didn't know that vcpkg automatically generates the link usage message (The package sdl2-gfx:x64-windows provides CMake targets:) which I thought was cool.

### CODE REVIEW

**Comment by dhkatz** [comment]
Seems like OSX CI has been broken for almost 2 weeks.

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Can you please correct the Version field in the CONTROL file? See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

**Comment by dhkatz** [comment]
Sorry about that, should be good now.

File: ports/sdl2-gfx/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -0,0 +1 @@
+_find_package(${ARGS})
```
Comment (JackBoosY) [inline ports/sdl2-gfx/vcpkg-cmake-wrapper.cmake]: If sdl2-gfx exports `config.cmake` / `target.cmake`, `vcpkg-cmake-wrapper` is no longer needed.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/sdl2-gfx/vcpkg-cmake-wrapper.cmake (hunk 1 above)
Comment (dhkatz) [reply ports/sdl2-gfx/vcpkg-cmake-wrapper.cmake]: Ah, I wasn't really sure. I sorta assumed I wouldn't need it but I was playing it safe. I will remove.

**Review by JackBoosY — APPROVED** [review]

File: ports/sdl2-gfx/CMakeLists.txt [hunk 2]
```diff
@@ -10,21 +12,42 @@ if(BUILD_SHARED_LIBS)
...
+  COMPATIBILITY AnyNewerVersion
+)
+
+install(EXPORT ${PROJECT_NAME}
+    FILE ${CONFIG_NAME}-config.cmake
+    NAMESPACE SDL2::
+    DESTINATION share/${CONFIG_NAME}/
+)
+
+export(PACKAGE sdl2-gfx)
```
Comment (dan-shaw) [inline ports/sdl2-gfx/CMakeLists.txt]: Is this line needed?

**Review by dan-shaw — APPROVED** [review]

**Review by ras0219-msft — APPROVED** [review]
Thanks for the PR! I improved a few small things while reviewing:
1. For targets provided by vcpkg, we do not use the CMake config version machinery
2. `export()` isn't needed -- that's only for using the binaries directly from the buildtrees
3. When built statically, consumers of this library will need to link SDL2 -- therefore, this `-config.cmake` file needed to transitively find `SDL2`.

**Comment by dhkatz** [comment]
Thank you everyone for the feedback, I will remember this for when I hopefully contribute more in the future.

**Comment by JackBoosY** [comment]
Could you remove `ignition-math4:x64-windows-static=fail` , `ignition-math4:x64-uwp=fail` and add `libqglviewer:x64-windows-static=fail` from _VCPKG_PATH/scritps/ci.baseline.txt_ ?
I will open a PR to fix libqglviewer build failure.

**Pull request merged**